<launch>

  <arg name="robot" default="orcvio_robot"/>
  <arg name="fixed_frame_id" default="world"/>

  <!-- bag parameters -->
  <arg name="path_bag" default="" />

  <arg name="visualize_rviz" default="true" />
  <node name="republish" type="republish" pkg="image_transport" output="screen" args="compressed in:=/camera/infra1/image_rect_raw  raw out:=/camera/infra1/image_rect_raw" />

  <!-- System Manager Nodelet  -->
  <group ns="$(arg robot)">
    <node pkg="nodelet" type="nodelet" name="system"
      args='standalone orcvio/SystemNodelet'
      output="screen">

      <param name="fixed_frame_id" value="$(arg fixed_frame_id)"/>
      <param name="child_frame_id" value="odom"/>

      <remap from="~imu" to="/camera/imu"/>
      <remap from="~cam0_image" to="/camera/infra1/image_rect_raw"/>

      <param name="config_file" value="$(find orcvio)/../../../config/rs_d435i.yaml"/>
      <param name="result_file" value="$(find orcvio)/../../../cache/temp_rmse.txt"/>

      <param name="load_groundtruth_flag" type="int" value="0"/>
      <!-- <param name="use_left_perturbation_flag" type="int" value="1" /> -->

    </node>
  </group>

    <!-- play the dataset -->
    <node pkg="rosbag" type="play" name="rosbag" args="$(arg path_bag)" required="true"/>
    
    <!-- open rviz  -->
    <group if="$(arg visualize_rviz)" >
      <node pkg="tf" type="static_transform_publisher" name="tf" args="0.0 0.0 0.0 0.0 0.0 0.0 1.0 world global 1000"/>
      <node type="rviz" name="rviz" pkg="rviz" args="-d $(find orcvio)/../../../ros_wrapper/src/orcvio/launch/rviz/orcvio_vio_rs_d435i.rviz" />
    </group>

</launch>
